contracts/core/adaptors/UniswapV3ViaRouterETH.sol

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.19;
+
+import {BaseUniRouter} from "./BaseUniRouter.sol";
+import {IAMM} from "../../interfaces/IAMM.sol";
+import {IUniswapV3} from "../../core/adaptors/IUniswapV3.sol";
+import {ISwapRouter} from "../../core/adaptors/IUniswapV3.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import "@uniswap/v3-periphery/contracts/libraries/TransferHelper.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+contract UniswapV3ViaRouterETH is BaseUniRouter, IAMM {
+    // Router https://docs.uniswap.org/contracts/v3/reference/deployments/base-deployments
+    address private constant wrappedETH = 0x4200000000000000000000000000000000000006;
+    address private intermidiateToken;
+
+    constructor() Ownable() {
+        fee = 10000;
+        intermidiateToken = wrappedETH;
+
+        // Approve the router to spend TOKEN.
+        TransferHelper.safeApprove(wrappedETH, router, type(uint256).max);
+    }
+
+    function setIntermidiateToken(address _intermidiateToken) external onlyOwner {
+        require(_intermidiateToken != wrappedETH, "UniswapV3ViaRouterETH: Invalid intermidiate token");
+        intermidiateToken = _intermidiateToken;
+
+        // Approve the router to spend TOKEN.
+        TransferHelper.safeApprove(intermidiateToken, router, type(uint256).max);
+    }
```

Comment:
If the intermediary token can only be WETH why add a set function ?

Reply:
Agreed.  Fixed.